build_tools/cmake/riscv.toolchain.cmake

```diff
@@ -55,6 +56,30 @@ if(RISCV_CPU STREQUAL "linux-riscv_64")
     "--iree-llvmcpu-target-cpu-features=+m,+a,+f,+d,+c,+zvl512b,+v"
     "--riscv-v-fixed-length-vector-lmul-max=8"
     CACHE INTERNAL "Default llvm codegen flags for testing purposes")
+elseif(RISCV_CPU STREQUAL "generic-riscv_64")
+  set(CMAKE_SYSTEM_PROCESSOR riscv64)
+  set(CMAKE_SYSTEM_NAME Generic)
+  set(CMAKE_CROSSCOMPILING ON CACHE BOOL "")
+  set(CMAKE_C_STANDARD 11)
+  set(CMAKE_C_EXTENSIONS OFF)     # Force the usage of _ISOC11_SOURCE
+  set(IREE_BUILD_BINDINGS_TFLITE OFF CACHE BOOL "" FORCE)
+  set(IREE_BUILD_BINDINGS_TFLITE_JAVA OFF CACHE BOOL "" FORCE)
+  set(IREE_HAL_DRIVER_DEFAULTS OFF CACHE BOOL "" FORCE)
+  set(IREE_HAL_DRIVER_LOCAL_SYNC ON CACHE BOOL "" FORCE)
+  set(IREE_HAL_EXECUTABLE_LOADER_DEFAULTS OFF CACHE BOOL "" FORCE)
+  set(IREE_HAL_EXECUTABLE_LOADER_EMBEDDED_ELF ON CACHE BOOL "" FORCE)
+  set(IREE_HAL_EXECUTABLE_LOADER_VMVX_MODULE ON CACHE BOOL "" FORCE)
+  set(IREE_HAL_EXECUTABLE_PLUGIN_DEFAULTS OFF CACHE BOOL "" FORCE)
+  set(IREE_HAL_EXECUTABLE_PLUGIN_EMBEDDED_ELF ON CACHE BOOL "" FORCE)
+  set(CMAKE_SYSTEM_LIBRARY_PATH "${RISCV_TOOLCHAIN_ROOT}/riscv64-unknown-elf/lib/")
+  set(IREE_ENABLE_THREADING OFF CACHE BOOL "" FORCE)
+  # Specify ISP spec for march=rv64gc. This is to resolve the mismatch between
+  # llvm and binutil ISA version.
+  set(RISCV_COMPILER_FLAGS "${RISCV_COMPILER_FLAGS} \
+      -march=rv64gc -mabi=lp64d -DIREE_PLATFORM_GENERIC=1 -DIREE_SYNCHRONIZATION_DISABLE_UNSAFE=1 \
```

Comment:
Please spell out the ISA version, or there will be version mismatch spamming during linking time.

Reply:
ahh... the ISA version is aligned by https://github.com/llvm/llvm-project/commit/dc90af501f00bb0bbbfde2d90360f074922e3e81. However, the toolchains that we provided in IREE project (and embedded in the docker) were built before this commit. It would still be better to specify the version to avoid confusion.

Reply:
OK, I'll add the ISA version.